library/aesce.c

```diff
@@ -17,6 +17,28 @@
  *  limitations under the License.
  */
 
+#if defined(__aarch64__) && !defined(__ARM_FEATURE_CRYPTO) && \
+    defined(__clang__) && __clang_major__ >= 4
+/* TODO: Re-consider above after https://reviews.llvm.org/D131064 merged.
+ *
+ * The intrinsic declaration are guarded by predefined ACLE macros in clang:
+ * these are normally only enabled by the -march option on the command line.
+ * By defining the macros ourselves we gain access to those declarations without
+ * requiring -march on the command line.
+ *
+ * `arm_neon.h` could be included by any header file, so we put these defines
+ * at the top of this file, before any includes.
+ */
+#define __ARM_FEATURE_CRYPTO 1
+/* See: https://arm-software.github.io/acle/main/acle.html#cryptographic-extensions
+ *
+ * `__ARM_FEATURE_CRYPTO` is deprecated, but we need to continue to specify it
+ * for older compilers.
+ */
+#define __ARM_FEATURE_AES    1
+#define MBEDTLS_NEED_TARGET_OPTIONS
```

Comment:
Non-blocker: I don't think this is a very informative name. Would something like `MBEDTLS_ENABLE_CRYPTO_EXTENSIONS_COMPILER_FLAG` be better?

Reply:
Since `MBEDTLS_NEED_TARGET_OPTIONS` does different things in different files, should it maybe be named differently? Or are we okay with the meaning "we need some (unspecified, but this file knows what it needs) target options applied later"?

Reply:
Agree, I missed that it isn't the same meaning. In which case I suggest changing to:

`MBEDTLS_ENABLE_ARM_CRYPTO_EXTENSIONS_COMPILER_FLAG` and `MBEDTLS_ENABLE_ARM_SHA3_EXTENSIONS_COMPILER_FLAG`

Reply:
Sounds good. And it might be bikeshedding, but "flags" are what you pass to the compiler on its command line; I prefer `MBEDTLS_SET_ARM_CRYPTO_EXTENSIONS_TARGET` and `MBEDTLS_SET_ARM_SHA3_EXTENSIONS_TARGET` (or `EXTENSION`, singular)

Reply:
OK, I prefer `FLAG` to `TARGET`, and I prefer just consistently using `EXTENSIONS` everywhere so that the names are more consistent, but no strong preference on either. I'm going to approve this, and I will raise a follow-up to tidy up the names.